front/src/applications/operationalStudies/components/Scenario/AddAndEditScenarioModal.tsx

```diff
@@ -0,0 +1,239 @@
+import ModalFooterSNCF from 'common/BootstrapSNCF/ModalSNCF/ModalFooterSNCF';
+import { ModalContext } from 'common/BootstrapSNCF/ModalSNCF/ModalProvider';
+import ChipsSNCF from 'common/BootstrapSNCF/ChipsSNCF';
+import { FaPencilAlt, FaPlus, FaTrash } from 'react-icons/fa';
+import { MdDescription, MdTitle } from 'react-icons/md';
+import InfraSelectorModal from 'common/InfraSelector/InfraSelectorModal';
+import { getInfraID, getProjectID, getStudyID } from 'reducers/osrdconf/selectors';
+import { useSelector, useDispatch } from 'react-redux';
+import { deleteRequest, patch, post } from 'common/requests';
+import { useNavigate } from 'react-router-dom';
+import { updateScenarioID } from 'reducers/osrdconf';
+import { setSuccess } from 'reducers/main';
+import { PROJECTS_URI, SCENARIOS_URI, STUDIES_URI } from '../operationalStudiesConsts';
+
+const configItemsDefaults = {
+  name: '',
+  description: '',
+  infra: undefined,
+  tags: [],
+};
+
+type configItemsTypes = {
+  id?: number;
+  name: string;
+  description: string;
+  infra: number | undefined;
+  tags: string[];
+};
+
+type Props = {
+  editionMode: false;
+  details?: configItemsTypes;
+  getScenarioDetail?: any;
+};
+
+export default function AddAndEditScenarioModal({
+  editionMode,
+  details,
+  getScenarioDetail,
+}: Props) {
+  const { t } = useTranslation('operationalStudies/scenario');
+  const { closeModal } = useContext(ModalContext);
+  const [configItems, setConfigItems] = useState<configItemsTypes>(details || configItemsDefaults);
+  const [displayErrors, setDisplayErrors] = useState(false);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const projectID = useSelector(getProjectID);
+  const studyID = useSelector(getStudyID);
+  const infraID = useSelector(getInfraID);
+
+  const removeTag = (idx: number) => {
+    const newTags: string[] = Array.from(configItems.tags);
+    newTags.splice(idx, 1);
+    setConfigItems({ ...configItems, tags: newTags });
+  };
+
+  const addTag = (tag: string) => {
+    const newTags: string[] = Array.from(configItems.tags);
+    newTags.push(tag);
+    setConfigItems({ ...configItems, tags: newTags });
+  };
+
+  const createScenario = async () => {
+    if (!configItems.name || !configItems.infra) {
+      setDisplayErrors(true);
+    } else {
+      try {
+        const result = await post(
+          `${PROJECTS_URI}${projectID}${STUDIES_URI}${studyID}${SCENARIOS_URI}`,
+          { ...configItems, infra: infraID }
+        );
+        dispatch(updateScenarioID(result.id));
+        navigate('/operational-studies/scenario');
+        closeModal();
+      } catch (error) {
+        console.error(error);
+      }
+    }
+  };
+
+  const modifyScenario = async () => {
+    if (!configItems.name) {
+      setDisplayErrors(true);
+    } else if (details) {
+      try {
+        await patch(
+          `${PROJECTS_URI}${projectID}${STUDIES_URI}${studyID}${SCENARIOS_URI}${details.id}/`,
```

Comment:
`${rootURI}${details.id}/`

Reply:
yes, and done too for study. replaced details by project/study/scenario everywhere